packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/blocs/onboarding_phone_bloc.dart

```diff
@@ -0,0 +1,134 @@
+import '../../base/models/errors/error_model.dart';
+import '../../base/models/user_model.dart';
+import '../models/country_code.dart';
+import '../models/onboarding_phone_errors.dart';
+import '../services/user_service.dart';
+
+part 'onboarding_phone_bloc.rxb.g.dart';
+part 'onboarding_phone_bloc_extensions.dart';
+
+/// A contract class containing all events of the OnboardingBloC
+abstract class OnboardingPhoneBlocEvents {
+  /// Sets the country code for the phone number
+  void setCountryCode(CountryCodeModel countryCode);
+
+  /// Sets the phone number
+  void setPhoneNumber(String phoneNumber);
+
+  /// Checks if the phone number is valid
+  void submitPhoneNumber();
+}
+
+/// A contract class containing all states of the OnboardingBloC.
+abstract class OnboardingPhoneBlocStates {
+  /// The loading state of the bloc
+  Stream<bool> get isLoading;
+
+  /// The user model
+  Stream<UserModel> get user;
+
+  /// The country code used for the phone number
+  Stream<CountryCodeModel?> get countryCode;
+
+  /// The phone number
+  Stream<String> get phoneNumber;
+
+  /// The state of the submit phone number button
+  Stream<bool> get submitPhoneNumberEnabled;
+
+  /// Error state of the bloc
+  Stream<ErrorModel> get errors;
+
+  /// Error state as a result of a validation
+  @RxBlocIgnoreState()
+  Stream<ErrorModel?> get validationErrors;
+}
+
+/// The [OnboardingPhoneBloc] handles the user's phone number setup
+@RxBloc()
+class OnboardingPhoneBloc extends $OnboardingPhoneBloc {
+  OnboardingPhoneBloc(this._userService) {
+    errorWithTagState
+        .map((errorWithTag) => errorWithTag.tag == tagValidation
+            ? errorWithTag.exception.asErrorModel()
+            : null)
+        .bind(_validationErrorSubject);
+  }
+
+  /// The user service used to communicate the user phone number
+  final UserService _userService;
+
+  /// Tag used to denote the submit phone number event
+  static const tagSubmitPhoneNumber = 'SubmitPhoneNumber';
+
+  /// Tag used to denote validation results
+  static const tagValidation = 'Validation';
+
+  /// Subject to handle validation errors
+  final _validationErrorSubject = BehaviorSubject<ErrorModel?>();
+
+  @override
+  Stream<ErrorModel> _mapToErrorsState() => errorWithTagState
+      .where((errorWithTag) => errorWithTag.tag == tagSubmitPhoneNumber)
+      .map((error) => error.exception)
+      .mapToErrorModel();
+
+  @override
+  Stream<ErrorModel?> get validationErrors => _validationErrorSubject;
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingWithTagState
+      .where((loadingWithTag) => loadingWithTag.tag == tagSubmitPhoneNumber)
+      .map((loadingWithTag) => loadingWithTag.loading)
+      .distinct();
+
+  @override
+  Stream<CountryCodeModel?> _mapToCountryCodeState() => _$setCountryCodeEvent;
+
+  @override
+  Stream<String> _mapToPhoneNumberState() => _$setPhoneNumberEvent
+      .map((phoneNumber) => phoneNumber.replaceAll(' ', ''))
```

Comment:
Avoid adding project specific code to the rx_bloc_cli. They are cases where the spaces in the phone are allowed/

Reply:
Fixed